client/dashboard/task-list/tasks/payments/methods.js

```diff
@@ -118,7 +118,10 @@ export function getPaymentMethods( {
 				</Fragment>
 			),
 			before: <WCPayIcon />,
-			visible: [ 'US' ].includes( countryCode ) && ! hasCbdIndustry,
+			visible:
+				[ 'US' ].includes( countryCode ) &&
+				! hasCbdIndustry &&
+				profileItems.jetpack_connected,
```

Comment:
We have a selector that performs this function:

https://github.com/woocommerce/woocommerce-admin/blob/7c0d4a34c6e213c990b16adb8ebc6b68d7a2e109/client/wc-api/onboarding/selectors.js#L95

Under the hood, it uses `wcSettings.dataEndpoints.jetpackStatus`.

Reply:
Ah interesting, thanks for that @psealock - I was the one who suggested the current approach :neckbeard:

Reply:
Hi @psealock, my first approach was using `isJetpackConnected` but later I changed my mind when I saw that it would be a cleaner solution to send the `jetpack connection` state in the `profileItems` param, that already has the `wccom_connected` value.
If you think that using `isJetpackConnected` is better I can go for it easily, I did this solution earlier :laughing:

Reply:
I do think it would be better than adding it to `profileItems` because when  https://github.com/woocommerce/woocommerce-admin/pull/4048/ gets merged, all the plugin information will be moved to its own data store.

> a cleaner solution to send the jetpack connection state in the profileItems param, that already has the wccom_connected value.

I hadn't noticed that `wccom_connected` was in profileItems 😄  . I think as part of #4048, `wccom_connected` should also be moved to the new plugins data store (or in a follow up PR).

For now, I think `isJetpackConnected` if you don't mind. That will make the refactor a bit easier. Thanks!

Reply:
Oh, now I see!
Good point @psealock
I'll go for it :slightly_smiling_face:
Thank you!

Reply:
Also that approach might clear up the failing test on CI:

1) WC_Tests_API_Onboarding_Profiles::test_schema

Failed asserting that actual size 16 matches expected size 15.

/tmp/wordpress/wp-content/plugins/woocommerce-admin/tests/api/onboarding-profile.php:102